Bowling game

alx1313 opened the pull request:
Added changes to previous bowling code

<!--- PR Title: What changed/implement in this PR -->
<!--- Be specific, don't write a generic title please --> 
 
## Description: Why did you change/implement that?
I added moss of this changes because the code needed to get better
Also some other changes were made because the code doesnt work as it should be
Its not finished yet, so i would make more changes

## Testing: Help me how can I test or look at the changes?
you can run the game on terminal writing: ruby bowling.rb
It will show the points of each shot,the scorer, and the total score of the game.
All the points i get are generated with "rand".

alx1313 committed:
Bowling game

Bowling code tested, with many improves to do

alx1313 commented:
Added bowling game code, rspec code(not finish yet), and rubycritic overview

EsmeSantoyo reviewed (commented):
You should divide your classes into different files and avoid code comments to make code more readable. You should avoid rubycritic files too because we don't review it. owo

jvargas98 commented on Gemfile:
```diff
@@ -0,0 +1,6 @@
+
+gem 'rspec'
+
+gem 'simplecov-small-badge', :require => false
```
remember add new line at the end of file

jvargas98 commented on bowling.rb:
```diff
@@ -0,0 +1,124 @@
+  end
+
+  def self.spare(rounds)
+    Scorer.multi_score(pins.pins)
+    @total_points += @score
+    @scorer[rounds] = @score
+    @multiplier += 1
+  end
+
+  def self.strike(rounds)
+    Scorer.multi_score(@balls.shot)
+    @total_points += @score
+    @scorer[rounds] = @score
+    @scorer[rounds + 1] = 0
+    @multiplier += 2
+    @balls.tries = 0
+    @balls.rounds += 1
+  end
+
+  def self.game(round)
+    Scorer.first_round(round)
+    @balls.shot == 10 ? Scorer.strike(round) : Scorer.second_round(round)
+    #puts @scorer[round]
+  end
+
+  def self.ingame(round)
+    if round == 19 && Pins.see_pins.zero?
+      Scorer.first_round(round)
+    else
+      Scorer.game(round)
+    end
+  end
+
+  def start_game
+    Scorer.new_game
+    [0, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20].each do |n|
+      n == 19 ? break : Scorer.ingame(n)
+    end
+  end
+end
```
I think it would be better if you separate the classes into different files as @EsmeSantoyo  said.

jvargas98 commented on spec/bowling_test_spec.rb:
```diff
@@ -0,0 +1,15 @@
+
+describe Ball do
+  describe 'new_shot' do
+    it 'Genera un numero, y lo guarda como puntuacion' do
```
maybe you can add more test cases

jvargas98 reviewed (commented):
please create a gitignore file to avoid uploading the rubycritic files

alx1313 committed:
Made changes on bowling game

Made changes on methods and manage better the data i generate
Added too more test to the other classes, but still unfinished
